### ISSUE TITLE
Update for chef-provisioning

### ISSUE THREAD

**Issue by bdupras** [vmtricks/chef-metal-vsphere#2]
This gem needs an update for chef-provisioning.  I'm soliciting PRs (and possibly maintainers) since I no longer have access to a vSphere infrastructure to dev against.

See also
RallySoftware-cookbooks/chef-metal-vsphere#2
RallySoftware-cookbooks/chef-metal-vsphere#3
chef/chef-provisioning#243

cc @jkeiser @dawsmith8 @derekschultz @grizo @MattMencel @rhass-r7 @mwrock

**Comment by rhass-r7** [comment]
If @jkeiser is interested and willing, it seems folding this into a provider for chef-provisioning-fog would be a way to hand-off and still provide people with the capability. I have tried to hack something together myself based on that, but I am having difficulty wrapping my head around the fog interfaces, specifically with the Vsphere driver.

**Comment by bdupras** [comment]
Yeah - I also spent a little time getting chef-metal-fog to work with vSphere, but after a couple days I decided that the layers of indirection were deeper than I wanted and so chef-metal-vsphere was born.  

The vmtricks version was written from scratch to take advantage of the [vmonkey](https://github.com/vmtricks/vmonkey) utility library (shameless plug), which cleaned up the code and increased overall test coverage.

I still think there's good reason to have a pure vSphere driver since there's a possibility of implementing control over a VM's target cluster, datastore, customization spec, vApp properties, and so on.  Let me know if you're interested in helping to get this driver modernized and released - I don't think it'd take long.

**Comment by aaron-lane** [comment]
@bdupras is it this org or the Rally one that contains the "true" chef-metal-vsphere gem?

**Comment by bdupras** [comment]
The Rally org code is the the gem `<= 0.2.0` as published on rubygems. The vmtricks code is the gem `= 0.3.beta.1` on rubygems.

**Comment by MattMencel** [comment]
Anyone coming to the Chef Community Summit on 3/31?  I was thinking of submitting this as an open space idea if there is more work to be done.

**Comment by bdupras** [comment]
I am not, though let me know if the open space happens and I'll try to be available for any q&a during the event.

**Comment by mwrock** [comment]
I will be there. My org, CenturyLink, uses a modified version of the original rally gem but its evolved quite a bit to support both windows and Linux and more network and storage scenarios.

As @bdupras knows I've been trying to get approval to open source the gem. Things have been picking up in that effort but I can't speculate when that will be finalized.

Whether this happens soon or not I'd love to discuss this topic and be involved as I can.

**Comment by MattMencel** [comment]
@bdupras What time zone are you at?  If you're over in Europe I can try to get a time slot that works better for you.  Earlier is better?

**Comment by bdupras** [comment]
US/Mountain - one hour ahead of the summit. I'm flexible most of the day
until 4pm Mountain, and then I'm unavailable the rest of the day.

**Comment by ewr** [comment]
I wasn't at yesterday's summit, but would love to hear if that session went anywhere. I'm also at ChefConf if anyone if around who wants to chat about provisioning on vsphere. I would be glad to jump in on trying to make this happen. Our environment is ~60 VMs on three ESXi hosts with a vcenter server.

**Comment by mwrock** [comment]
Hey Eric,

I'm happy to discuss and am here. oday is bad but maybe sometime tomorrow? Any pre conf, break time or lunch works for me.

**Comment by ewr** [comment]
Cool. Just dropped you an email with my cell and suggesting breakfast as a time.

**Comment by MattMencel** [comment]
I chatted with the VMware rep last night.  He says that they are open
sourcing a Chef provisioner soon.  Maybe take that with a grain of salt but
it sounded interesting.  Eric can you send me an email too?  I would be
interested in chatting about it too.

**Comment by ewr** [comment]
Cool. Sent you an email with my cell so that we can coordinate meeting up at breakfast.

I stopped by the VMware booth, but it was just cloud people without much to say about vSphere. I'm a little guy... I don't get the fun tools for cloud provisioning.

**Comment by aaron-lane** [comment]
@mwrock I see clc-chef-metal-vsphere on RubyGems. Is that CL's version of this gem?

**Comment by bdupras** [comment]
Yep

**Comment by mwrock** [comment]
FYI: We just opened up our driver that runs against chef-provisioning v1.1.1:

https://github.com/CenturyLinkCloud/chef-provisioning-vsphere
ruby gems: https://rubygems.org/gems/chef-provisioning-vsphere

I'll be updating the readme and preparing a CenturyLink blog post today likely to be published next week.
